### ISSUE TITLE
Demos with GLFW3

### ISSUE THREAD

**Issue by utgarda** [slembcke/Chipmunk2D#115]
For some time already my system has GLFW 3 by default, and that's the only version I get from the repos, and the demo app doesn't compile with it. I'm not even sure how to downgrade, that's how far from being a C programmer I am. Still, I tried something: 
https://github.com/utgarda/Chipmunk2D/commit/63731291c876ecba5be9689647aee3b1fa480d7d

I got it compiled and running with GLFW3, most of the API changes are about multi-window mode, but still it doesn't work as expected - I see some shapes moving, but colors and scale are wrong, and no response to keys pressed. 

Is it possible to migrate demos to glfw3, and is it a good idea?

**Comment by slembcke** [comment]
I considered doing it a year or so ago. I forget what the problem I ran into was exactly. Maybe it was that the API was different enough to be annoying. It was also probably a mistake to embed the Mac/Win library binaries in the repo to begin with. IIRC the binaries for GLFW3 were much larger. I decided it was easier to just stick with whatever versions I already had. The GL code in the demo project is really not very good, and I didn't really want to take the time to fix up the demos when they worked good enough for the 10 minutes that they needed to run for.

I'll look into it again soon though.

**Comment by utgarda** [comment]
@slembcke thanks a bunch! The demo app code has a warning not to use it as a GL tutorial, so it's cool, it can have pretty much any level of quality, as long as it allows one to see Chipmunk in action as soon as it's downloaded and built. I am currently trying to write another bindings library in Java using JNA - an easy way, but some parts are awfully slow compared to C code or event to JNI bindings. So I'm porting the examples, and it would be really helpful to launch the examples alongside to visually compare performance at a glance, without proper benchmarking. https://github.com/utgarda/chipmunk2d-jna/ - just a start, but LogoSmash example actually works.
Many years ago I tried to port these demos to Ruby, but that didn't go far. But since that time I went into gamedev, and would like to use Chipmunk instead of Box2d, that's my reason to create Java bindings with a more permissive license than https://github.com/johang/chipmunk-for-java. If JNA turns to be not-so-usable for this, I'll turn to JNI and take https://github.com/libgdx/libgdx/tree/master/extensions/gdx-box2d/gdx-box2d as an example.
